execution-core/src/transfer.rs

```diff
@@ -7,600 +7,260 @@
 //! Types related to Dusk's transfer contract that are shared across the
 //! network.
 
-use alloc::string::String;
 use alloc::vec::Vec;
 
-use core::cmp;
-
 use bytecheck::CheckBytes;
-use dusk_bytes::{DeserializableSlice, Error as BytesError, Serializable};
-use dusk_poseidon::{Domain, Hash};
-use ff::Field;
-use rand::{CryptoRng, RngCore};
-use rkyv::{
-    ser::serializers::AllocSerializer, Archive, Deserialize, Fallible,
-    Serialize,
-};
+use dusk_bytes::{DeserializableSlice, Error as BytesError};
+use rkyv::{Archive, Deserialize, Serialize};
 
 use crate::{
-    BlsPublicKey, BlsScalar, BlsSecretKey, BlsSignature, JubJubScalar, Note,
-    PublicKey, SchnorrSecretKey, SchnorrSignature, Sender, StealthAddress,
-};
-
-mod transaction;
-pub use transaction::{
-    MoonlightPayload, MoonlightTransaction, PhoenixPayload, PhoenixTransaction,
-    Transaction,
+    signatures::bls::{
+        PublicKey as AccountPublicKey, Signature as AccountSignature,
+    },
+    BlsScalar, ContractId,
 };
 
-use crate::bytecode::Bytecode;
-use crate::reader::{read_arr, read_str, read_vec};
-
-/// Unique ID to identify a contract.
-pub type ContractId = [u8; 32];
+pub mod contract_exec;
```

Comment:
Might we find a better name here? I think even just `contract` might be better.

Reply:
Although I'm also not entirely happy with the name `contract_exec`, I'm afraid that simply calling it `contract` is going to be too misleading: the module path `transfer::contract` would in my mind need to contain all the types associated with the transfer-contract. But this module is specifically for everything associated with the contract execution functionalities of the transfer-contract (i.e. contract-calls and contract-deployment).

Reply:
I also really don't like `contract_exec`, but since I can't find a better name, I'm also not going to hold this up on account of it.